src/api/users.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{models, error, Octocrab, Result};
+
+pub struct CurrentAuthHandler<'octo> {
+    crab: &'octo Octocrab,
+}
+
+impl<'octo> CurrentAuthHandler<'octo> {
+    pub(crate) fn new(crab: &'octo Octocrab) -> Self {
+        Self { crab }
+    }
+
+    /// Fetches information about the current user.
+    /// If the current client is unauthorized, this returns an `Ok(None)` value.
+    pub async fn user(&self) -> Result<Option<models::User>> {
```

Comment:
I'd remove the error path, and change the return type to `Result<models::User>`. I'd consider it programmer error to use the `CurrentAuthHandler` when using a token that doesn't have an authenticated user behind. And people who need to handle it will get an error anyway.